charts/redpanda/templates/_statefulset.tpl

```diff
@@ -0,0 +1,46 @@
+{{/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/}}
+
+{{- define "statefulset-pod-labels" -}}
+{{- /*
+  StatefulSets cannot change their selector. Use the existing one even if it's broken.
+  New installs will get better selectors.
+*/ -}}
+{{- $sts := lookup "apps/v1" "StatefulSet" .Release.Namespace (include "redpanda.fullname" .) -}}
```

Comment:
I'm starting to question aren't we too clever in templating

Reply:
I do agree, @RafalKorepta. But I want to avoid completely breaking existing users since we just did that back in September. I think this is the only way to do that.